### PR TITLE
2565 Clinician merging: encounter

### LINKED ISSUE #2565: Clinician merging: encounter
## Is your feature request related to a problem? Please describe 👀

We need to replace all FKs to `clinician.id` with FKs to `clinician_link.id` instead, to allow easy merge management.

## Describe the solution you'd like 🎁

Base: [feature/merging](https://github.com/msupply-foundation/open-msupply/tree/feature/merging)
Using the PR for [clinician_link + clinician_store_join](https://github.com/msupply-foundation/open-msupply/issues/2454) as a guide:
- [ ] Add optional `encounter.clinician_link_id` and migrate with `encounter.clinician_link_id = encounter.clinician_id`
- [ ] Remove `encounter.clinician_id`
- [ ] Update any views that relied on relations of `encounter.clinician_id`
- [ ] Update `encounter` at the repository layer to return related optional `clinician` records based on `encounter.clinician_link_id` rather than `encounter.clinician_id`

### Moneyworks Jobcode 🧰

OMS:MERGE

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #2565

# 👩🏻‍💻 What does this PR do? 
- Adds clinician_link_id to encounter 

# 🧪 How has/should this change been tested? 
- Make sure tests still work

### CODE REVIEW

**Review by Chris-Petty — CHANGES_REQUESTED** [review]

File: server/repository/src/migrations/v1_06_00/encounter_add_clinician_link_id.rs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    #[cfg(feature = "postgres")]
+    sql!(
+        connection,
+        r#"
+            ALTER TABLE encounter
+            ADD COLUMN clinician_link_id TEXT;
+
+            UPDATE encounter SET clinician_link_id = clinician_id;
+
+            ALTER TABLE encounter ADD CONSTRAINT encounter_clinician_link_id_fkey FOREIGN KEY (clinician_link_id) REFERENCES clinician_link(id);
```
Comment (andreievg) [inline server/repository/src/migrations/v1_06_00/encounter_add_clinician_link_id.rs:13]: do we need to add index here ?

**Review by andreievg — APPROVED** [review]

File: server/repository/src/migrations/v1_06_00/encounter_add_clinician_link_id.rs (hunk 1 above)
Comment (Chris-Petty) [reply server/repository/src/migrations/v1_06_00/encounter_add_clinician_link_id.rs:13]: Moving index sql from the not postgres block to the general block at the bottom, will do for the other migrations too!

Comment (Chris-Petty) [reply server/repository/src/migrations/v1_06_00/encounter_add_clinician_link_id.rs:13]: On second thought I'll look tomorrow - should check the FK constraint syntax for pgsql doesn't automatically generate the index. Though if it doesn't then surely we don't need this to run for postgres either, or at least all the ones for FKs. 

https://github.com/msupply-foundation/open-msupply/blob/a025f2405bf5e9d7b6f4790b080fcdf571ab83ef/server/repository/src/migrations/v1_06_00/indexes.rs#L3-L8

Comment (Chris-Petty) [reply server/repository/src/migrations/v1_06_00/encounter_add_clinician_link_id.rs:13]: It doesn't. I'm noticing a bunch of mistakes like this across the migrations for merge, and possibly other mistakes too. I'll fix this one here but open an issue for the rest.

**Pull request merged**